arrow-core-data/src/main/kotlin/arrow/core/Ior.kt

```diff
@@ -277,28 +272,42 @@ sealed class Ior<out A, out B> : IorOf<A, B> {
       i: Ior<A, I>,
       j: Ior<A, J>,
       k: Ior<A, K>,
-      crossinline map: (B, C, D, E, F, G, H, I, J, K) -> L
+      map: (B, C, D, E, F, G, H, I, J, K) -> L
     ): Ior<A, L> =
-      b.flatMap1(SA) { bb ->
-        c.flatMap1(SA) { cc ->
-          d.flatMap1(SA) { dd ->
-            e.flatMap1(SA) { ee ->
-              f.flatMap1(SA) { ff ->
-                g.flatMap1(SA) { gg ->
-                  h.flatMap1(SA) { hh ->
-                    i.flatMap1(SA) { ii ->
-                      j.flatMap1(SA) { jj ->
-                        k.map { kk ->
-                          map(bb, cc, dd, ee, ff, gg, hh, ii, jj, kk)
-                        }
-                      }
-                    }
-                  }
-                }
-              }
-            }
-          }
+      if (b is Right && c is Right && d is Right && e is Right && f is Right && g is Right && h is Right && i is Right && j is Right && k is Right) {
```

Comment:
Awesome! This should be much faster than the derived implementation from `Apply`, and this bytecode should also be much shorter than the `flatMap` derived version.